### ISSUE TITLE
Training model produce all metrics to 0.0

### ISSUE THREAD

**Issue by victorc365** [tensorflow/recommenders#191]
Hi I'm using the model presented in the retrieval [tutorial](https://www.tensorflow.org/recommenders/examples/basic_retrieval), with another dataset but when I train the model all metrics stay at zero and nothing improve. Has anyone else had the same issue? Why is it happening? How to solve?

**Comment by LAANAITI** [comment]
Hi, if you have executed the code as it is, it is normal, because you have to set training = True when you calculate the loss function (compute_loss) in MovieLensModel class.

**Comment by anisayari** [comment]
Just replace `self.task(user_embeddings, positive_movie_embeddings)` by `self.task(user_embeddings, positive_movie_embeddings, compute_metrics= not training)`

**Comment by juliobguedes** [comment]
Yes, this seems to be a real issue. Please check #119.

The answer given by @anisayari does not solve the problem but speeds up training by not computing the metrics when training, as I stated in the issue. @maciejkula can you confirm my understanding?

**Comment by maciejkula** [comment]
That's correct. However, test metrics should not be zero - if they are, this suggests a problem with the model.
